buildenv/jenkins/copyrightCheck

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2017 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+
+Boolean FAIL = false
+String SRC_REPO = 'https://github.com/eclipse/openj9.git'
+def BAD_FILES = []
+String HASHES = '###################################'
+
+stage('Copyright Check') {
+    node ('master') {
+        timestamps {
+            git url: SRC_REPO
+            sh "git fetch --tags --progress origin +refs/pull/*:refs/remotes/origin/pr/*"
+            sh "git checkout --detach ${sha1}"
+            sh 'git fetch origin'
+            FILES = sh (
+                script: "git diff --diff-filter=ACM --name-only origin/${ghprbTargetBranch} HEAD",
+                returnStdout: true
+            ).trim()
+            echo FILES
+            def FILES_LIST = FILES.split("\\r?\\n")
+            DATE_YEAR = sh (
+                script: "date +%Y",
+                returnStdout: true
+            ).trim()
+            FILES_LIST.each() {
+                println "Checking file: '${it}'"
+                RESULT = sh (
+                    script: "grep -qE \"Copyright \\(c\\) ([0-9]{4}), ${DATE_YEAR} IBM Corp. and others\" '${it}'",
```

Comment:
If you use single quotes around the pattern, I expect fewer escapes would be required.

Reply:
updated commit, seems to still work properly

Reply:
I suspect you didn't have `\` escapes before, but it seems right now.